### DISCUSSION TITLE
Passing an array to category for dynamic categories

### DISCUSSION THREAD

**Discussion by jonwinstanley** [spatie/laravel-feed#191] [Ideas]
I'd like to pass a small number of article tags to the category method and have them rendered as <category /> elements in the feed.

However, an array is not permitted, and neither is a comma separated list. The category method requires additional arguments to be passed.

With a dynamic number of tags, this means I have chosen to do as follows, which seems inefficient.

```
            ->category(
                isset($tags[0]) ? $tags[0]->name : 'News',
                isset($tags[1]) ? $tags[1]->name : '',
                isset($tags[2]) ? $tags[2]->name : ''
            )
```

Then in my view, I exclude empty string categories.

```
                @foreach($item->category as $category)
                    @if ($category)
                    <category>{{ $category }}</category>
                    @endif
                @endforeach

```

Am I missing something here, is having dynamic tags a bad idea for some reason?

Thanks!

**Comment by julienmru** [comment]
Hello : )

+1 that would be something very helpful to me as I have a dynamic number of categories to assign to an item.

Right now, I'm considering playing with [`call_user_func_array`](https://www.php.net/manual/en/function.call-user-func-array.php) but that's not as neat as it could be.

Thanks!
